lib/shopify_api/resources/discount_code.rb

```diff
@@ -5,5 +5,15 @@ class DiscountCode < Base
     def price_rule_id
       @prefix_options[:price_rule_id]
     end
+
+    def self.lookup(code)
+      find(:one, from: '/admin/discount_codes/lookup.json?code=' + code)
+    rescue ActiveResource::Redirection => ex
+      r = find(:one, from: ex.response['Location'])
```

Comment:
if we store `ex.response['Location']` to a variable, there will be a little less repetition here

Reply:
good idea, made the change.